### PR TITLE
Add OpenRA and OS language to exception messages

### PR DESCRIPTION
**Pull request by Mailaender**
to cover incidents like https://github.com/OpenRA/OpenRA/issues/19996.

### CODE REVIEW

File: OpenRA.Game/Support/ExceptionHandler.cs [hunk 1]
```diff
@@ -24,25 +24,29 @@ public static void HandleFatalError(Exception ex)
...
+				Log.Write("exception", $"on map {map.Uid} ({map.Title} by {map.Author}).");
 			}
 
-			Log.Write("exception", "Date: {0:u}", DateTime.UtcNow);
-			Log.Write("exception", "Operating System: {0} ({1})", Platform.CurrentPlatform, Environment.OSVersion);
-			Log.Write("exception", "Runtime Version: {0}", Platform.RuntimeVersion);
+			Log.Write("exception", $"Date: {DateTime.UtcNow:u}");
+			Log.Write("exception", $"Operating System: {Platform.CurrentPlatform} ({Environment.OSVersion})");
+			Log.Write("exception", $"Runtime Version: {Platform.RuntimeVersion}", Platform.RuntimeVersion);
+			Log.Write("exception", $"System Language: {CultureInfo.InstalledUICulture} (Installed) {CultureInfo.CurrentCulture} (Current) {CultureInfo.CurrentUICulture} (Current UI)");
```
Comment (PunkPun) [inline OpenRA.Game/Support/ExceptionHandler.cs]: What is this line for? None of the variables show up for me

Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: It shows `de_DE` for me. https://learn.microsoft.com/en-US/dotnet/api/system.globalization.cultureinfo.installeduiculture?view=net-6.0 It is required for debugging crashes that happen on language English, system language Bulgarian like the one I linked above. Different decimal separators etc.

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: `de_DE` for all variables? On my end I only see the words in the brackets what confused me. Shouldn't there be some defaults?

Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: Yes, for all. I don't know why it doesn't work on Mac.

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: It appears it is reading it as iv (Invariant Country)

Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: What locale is your system set to?

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: It's UK english, using US keyboard. I tried to change it to other languages and it still didn't pick up

Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: Are you running @mono? https://github.com/dotnet/runtime/issues/27159

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: No

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: For this PR I think adding a default value is enough.

**Review by abcdefg30 — DISMISSED** [review]
Works on Windows.

File: OpenRA.Game/Support/ExceptionHandler.cs (hunk 1 above)
Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: Is the value null on Mac?

Comment (Mailaender) [reply OpenRA.Game/Support/ExceptionHandler.cs]: What would be the default?

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: It's not null, all the variables have values. They return a shorthand IC (Invariant Country), but when you ask for their string directly like you currently do it returns an empty string (or perhaps null cause that's the same thing)

File: OpenRA.Game/Support/ExceptionHandler.cs [hunk 2]
```diff
@@ -24,25 +24,30 @@ public static void HandleFatalError(Exception ex)
...
+				Log.Write("exception", $"on map {map.Uid} ({map.Title} by {map.Author}).");
 			}
 
-			Log.Write("exception", "Date: {0:u}", DateTime.UtcNow);
-			Log.Write("exception", "Operating System: {0} ({1})", Platform.CurrentPlatform, Environment.OSVersion);
-			Log.Write("exception", "Runtime Version: {0}", Platform.RuntimeVersion);
+			Log.Write("exception", $"Date: {DateTime.UtcNow:u}");
+			Log.Write("exception", $"Operating System: {Platform.CurrentPlatform} ({Environment.OSVersion})");
+			Log.Write("exception", $"Runtime Version: {Platform.RuntimeVersion}", Platform.RuntimeVersion);
+			Log.Write("exception", $"Installed Language: {CultureInfo.InstalledUICulture.Name} (Installed) {CultureInfo.CurrentCulture.Name} (Current) {CultureInfo.CurrentUICulture.Name} (Current UI)");
```
Comment (PunkPun) [inline OpenRA.Game/Support/ExceptionHandler.cs]: Invariant Culture apparently doesn't have a name

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: I'm still getting 
```Installed Language:  (Installed)  (Current)  (Current UI)```

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: `.TwoLetterISOLanguageName` returns iv
`.ThreeLetterISOLanguageName` returns ivl

Comment (PunkPun) [reply OpenRA.Game/Support/ExceptionHandler.cs]: `.NativeName` returns `Invariant Language (Invariant Country)`

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fa6d48b92c611e879f87350ea3c504bfc31460bc...b9f14d08e683fc353d1936a2e1d8bad91e02d0e1)

**Pull request merged**